File: Assignment5/Coin.java
import java.awt.Graphics;
import java.awt.image.BufferedImage;

public class Coin extends Sprite 
{
    double xVelocity, yVelocity;
    static BufferedImage coinImage = null;
    Coin(Sprite s)
    {
        this.x = s.x + (this.w/2);
        this.y = s.y - s.h;
        this.w = 100;
        this.h = 100;
        this.isCoinBlock = false;
        this.xVelocity = Math.random() * 10;
        this.yVelocity = -(Math.random() * 10);

        this.LoadImage();
    }

    void LoadImage()
    {
        if(coinImage == null)
            coinImage = View.loadImage("coin.png");
    }

    void update() 
    {
        yVelocity += 0.5;
        y += yVelocity;
        x += xVelocity;
    }

    void draw(Graphics g) 
    {
        g.drawImage(coinImage, this.x - Brick.m.cameraPos, this.y, this.w, this.h, null);
    }

    Json Marshal() 
    {
        return Json.newObject();
    }

    @Override
    public boolean isCoin()
    {
        return true;
    }
}

File: Assignment7/Coin.js
class Coin extends Sprite {
    constructor(x,y,w,h,model) {
        super (x,y,w,h,model)
        // create random horizontal value for coin 
        this.dy = Math.floor(Math.random() * 10) + 1;

        // setup image
        let tempImage = new Image();
        tempImage.src = "coin.png";
        super.image = tempImage;
    }

    // remove coin if goes below map, add random horizontal and constant vertical values each update
    update() {
        if(this.y > 550) {
            game.model.sprites.splice(game.model.sprites.indexOf(this), 1);
        } else {
            this.x += this.dy;
            this.y += 5;
        }
    }

    // draw the coin
    draw(ctx) {
        ctx.drawImage(this.image, this.x - game.model.cameraPos, this.y, 50, 50);
    }
}